### PR TITLE
Added missing query param and updated status enums

### PR DESCRIPTION
**Pull request by saisankargochhayat**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [ ] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [x] Other, please clarify:
    - The Resource Bridge swagger definition was missing a query param on the listkeys route, which is fixed with this PR
And also updated the set of statuses to be updated. 

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

File: specification/resourceconnector/resource-manager/Microsoft.ResourceConnector/stable/2022-10-27/appliances.json [hunk 1]
```diff
@@ -453,6 +453,12 @@
           },
           {
             "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "name": "artifactType",
+            "in": "query",
+            "type": "string",
+            "description": "This sets the type of artifact being returned, when empty no artifact endpoint is returned."
           }
```
Comment (rkmanda) [inline specification/resourceconnector/resource-manager/Microsoft.ResourceConnector/stable/2022-10-27/appliances.json:462]: send this in the payload not as a query parameter

Comment (amanohar) [reply specification/resourceconnector/resource-manager/Microsoft.ResourceConnector/stable/2022-10-27/appliances.json:462]: @rkmanda adding context the ask for change GET to POST for listclusterusercredentials API in AKS at the time came from ARM engineering team.

Comment (rkmanda) [reply specification/resourceconnector/resource-manager/Microsoft.ResourceConnector/stable/2022-10-27/appliances.json:462]: The problem is not with the POST action but with having a query parameter. I dont quite follow the challenge with including the parameter in the request payload.

https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#:~:text=All%20parameters%20for%20the%20action%20request%20should%20be%20contained%20in%20the%20request%20body%20%E2%80%93%20and%20not%20in%20the%20URL.%20This%20matches%20the%20OData%20and%20Azure%20REST%20guidelines%20and%20will%20avoid%20impacting%20the%20authZ%20checks.

Comment (rkmanda) [reply specification/resourceconnector/resource-manager/Microsoft.ResourceConnector/stable/2022-10-27/appliances.json:462]: Approving as on-off implementation. For future please follow the guideline as explained

Here are the drawbacks to putting something on the URI. For these reasons we strongly recommend placing the parameters in the body of the request. If we encounter a strong justification for why the pattern cannot be followed, we are open to providing an exception 

- You have to deal with URI encoding which could munge up the data
- Intermediary services may have to worry about sanitizing the data on the URI during logging.  This is easy for well known patterns like SAS URIs but not easy for arbitrary query params
- You won't be able to use linked access checks for any content in query params (but you could if the content is in the body)
- The URI can only hold so much content.  If you put something on the query string and then realize users need to pass multiples of it or the content becomes long you may be stuck with a big breaking change to pull it into the request body

**Review by rkmanda — APPROVED** [review]
:shipit:

**Comment by rkmanda** [comment]
azp /run

File: specification/resourceconnector/resource-manager/Microsoft.ResourceConnector/stable/2022-10-27/appliances.json (hunk 1 above)
Comment (saisankargochhayat) [reply specification/resourceconnector/resource-manager/Microsoft.ResourceConnector/stable/2022-10-27/appliances.json:462]: Thanks for the feedback and discussion offline

**Comment by saisankargochhayat** [comment]
azp /run

**Pull request merged**